# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [2]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [3]:
np.random.seed(50)
series1 = pd.Series(np.random.randint(1, 5, size=100))
series2 = pd.Series(np.random.randint(1, 4, size=100))
series3 = pd.Series(np.random.randint(10000, 30001, size=100))

### Step 3. Let's create a DataFrame by joinning the Series by column

In [4]:
df = pd.concat([series1, series2, series3], axis=1)
df

0  1      2
0   1  1  19091
1   1  1  29484
2   4  1  27083
3   2  1  16788
4   2  2  21945
.. .. ..    ...
95  2  3  29155
96  2  1  18959
97  1  1  23294
98  3  2  15796
99  2  3  23696

[100 rows x 3 columns]

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [5]:
df = df.rename(columns= {
     0 : 'bedrs',
     1 : 'bathrs',
     2 : 'price_sqr_meter'
})
df

bedrs  bathrs  price_sqr_meter
0       1       1            19091
1       1       1            29484
2       4       1            27083
3       2       1            16788
4       2       2            21945
..    ...     ...              ...
95      2       3            29155
96      2       1            18959
97      1       1            23294
98      3       2            15796
99      2       3            23696

[100 rows x 3 columns]

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [6]:
bigcolumn = pd.concat([series1, series2, series3], axis=0, ignore_index=True)
df_bigcolumn = pd.DataFrame({'bigcolumn': bigcolumn})
df_bigcolumn

bigcolumn
0            1
1            1
2            4
3            2
4            2
..         ...
295      29155
296      18959
297      23294
298      15796
299      23696

[300 rows x 1 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [7]:
# NO it will continue after 99
len(df_bigcolumn)

300

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [8]:
bigcolumn.reset_index(drop=True, inplace=True)
bigcolumn

0          1
1          1
2          4
3          2
4          2
       ...  
295    29155
296    18959
297    23294
298    15796
299    23696
Length: 300, dtype: int32